src/student/student.service.ts

```diff
@@ -134,200 +135,164 @@ export class StudentService {
-        throw new NotAcceptableException(
-          `Użytkownik z adresem e-mail: ${email} już istnieje w bazie`,
-        );
-      }
+            const update2 = (email)
+                ?
+                await this.userRepository.createQueryBuilder()
+                    .update('users')
+                    .set({
+                        email,   //updatedAt: () => 'CURRENT_TIMESTAMP',
+                    })
+                    .where('id = :id', {id: student.userId})
+                    .execute()
+                :
+                console.log('no [users] rows affected');
+            //console.log(update2)
 
-      const update1 =
-        !Object.values({ ...restOfDetails, githubName }).every(
-          (detail) => detail === undefined,
-        ) || githubName
-          ? await this.studentRepository
-              .createQueryBuilder()
-              .update('students')
-              .set({
-                githubName,
-                ...restOfDetails,
-                updatedAt: dataUpdatedAt,
-              })
-              .where('id = :studentId', { studentId })
-              .execute()
-          : console.log('no [students] rows affected');
-      //console.log(update1)
+            const urlsUpdate = this.projectRepository.createQueryBuilder()
+                .update('projects')
+            if (studentProfileDetails.bonusProjectUrl && studentProfileDetails.portfolioUrl) {
+                urlsUpdate
+                    .set(setBonusProjectUrl)
+                    .where('projects.type = :type', {type: projectTypeEnum.bonusProject})
+                    .andWhere('projects.user_id = :id', {id: student.userId})
+                await urlsUpdate.execute()
+                urlsUpdate
+                    .set(setPortfolioUrl)
+                    .where('projects.type = :type', {type: projectTypeEnum.portfolio})
+            } else if (studentProfileDetails.bonusProjectUrl) {
+                urlsUpdate
+                    .set(setBonusProjectUrl)
+                    .where('projects.type = :type', {type: projectTypeEnum.bonusProject})
+            } else if (studentProfileDetails.portfolioUrl) {
+                urlsUpdate
+                    .set(setPortfolioUrl)
+                    .where('projects.type = :type', {type: projectTypeEnum.portfolio})
+            } else {
+                console.log('no [projects] rows affected');
+                return {
+                    isSuccess: !!(update1 as UpdateResult || update2 as UpdateResult),
+                    message: (update1 as UpdateResult || update2 as UpdateResult) ? 'All affected records' +
+                        ' have been successfully updated' : 'no [students/users/projects] rows affected'
+                }
+            }
+            urlsUpdate.andWhere('projects.user_id = :id', {id: student.userId})
 
-      const update2 =
-        email && uniqueUserMail.length === 0
-          ? await this.userRepository
```

Comment:
Tutaj ponownie odnosimy się do userRepository, trzeba to zmienić aby odnosić się bezpośrednio do userService

Reply:
jak wyżej :)